File: 2-strings/java/344-reverse-string.java
public class Solution{
    public static void reverseString(char[] s) {
        for(int i=0; i<s.length/2; i++){
            char tmp = s[i];
            s[i] = s[s.length-1-i];
            s[s.length-1-i] = tmp;
        }
    }

    public static void main(String args[]){
        char[] s = {'h', 'e','l','l','o'};
        reverseString(s);

        for(int i=0; i<s.length; i++){
            System.out.print(s[i]);
        }
        System.out.println("");
    }
}

// Time Complexity: O(n)
// Space Complexity: O(1)

File: 2-strings/cpp/344-reverse-string.cpp
#include <bits/stdc++.h>
using namespace std;

void reverseString(vector<char>& s) {
    for(int i=0; i<s.size()/2; i++){
        swap(s[i], s[s.size()-1-i]);
    }
}

int main(){
    vector<char> s = {'h','e','l','l','o'};
    reverseString(s);

    for(auto i: s){
        cout << i << " ";
    }
    cout << endl;
    return 0;
}

// Time Complexity: O(n)
// Space Complexity: O(1)
